crates/starlight/src/vm.rs

```diff
@@ -818,10 +867,11 @@ pub struct GlobalData {
     pub(crate) map_prototype: Option<GcPointer<JsObject>>,
     pub(crate) set_prototype: Option<GcPointer<JsObject>>,
     pub(crate) regexp_structure: Option<GcPointer<Structure>>,
-    pub(crate) regexp_object: Option<GcPointer<JsObject>>,
+    pub(crate) regexp_prototype: Option<GcPointer<JsObject>>,
     pub(crate) array_buffer_prototype: Option<GcPointer<JsObject>>,
     pub(crate) array_buffer_structure: Option<GcPointer<Structure>>,
     pub(crate) data_view_structure: Option<GcPointer<Structure>>,
+    pub(crate) data_view_prototype: Option<GcPointer<JsObject>>,
```

Comment:
Is it necessary to save data_view and regexp prototypes there?

Reply:
Not necessary. I think it's ok to save data_view and regexp prototypes? They can also be snapshotted like other prototypes. What do you think?

Reply:
thing is `data_view_structure` already stores its prototype and serialization routine will serialize this prototype too and data_view_prototype with regexp_prototype not required out there

Reply:
That makes sense. And I think it's also ok to serialize data_view_prototype and regexp_prototype. And it will only serialize two more GcPointer. On the other hand, one main advantage I think is that we can use the same logic to get a prototype or constructor from global_data.